src/hotspot/share/runtime/threadSMR.cpp

```diff
@@ -466,9 +466,14 @@ void SafeThreadsListPtr::acquire_stable_list_nested_path() {
   if (EnableThreadSMRStatistics) {
     _thread->inc_nested_threads_hazard_ptr_cnt();
   }
-  current_list->inc_nested_handle_cnt();
-  _previous->_has_ref_count = true;  // promote SafeThreadsListPtr to be reference counted
-  _thread->_threads_hazard_ptr = NULL;  // clear the hazard ptr so we can go through the fast path below
+  if (!_previous->_has_ref_count) {
+    // Promote the thread's current SafeThreadsListPtr to be reference counted.
+    current_list->inc_nested_handle_cnt();
+    _previous->_has_ref_count = true;
+  }
+  // Clear the hazard ptr so we can go through the fast path below and
+  // acquire a nested stable ThreadsList.
+  _thread->_threads_hazard_ptr = NULL;
```

Comment:
Should be Atomic::store()

Reply:
That's a bug in the baseline also, but I can fix it since I'm already
touching that line.

I also noticed that we weren't really consistent about using atomic
operations with the _threads_hazard_ptr field and we don't always
use the (mis-named) get_threads_hazard_ptr(). I plan to make
another cleanup pass with a different bug ID for those.